gui/wxpython/dbmgr/base.py

```diff
@@ -220,155 +220,163 @@ def LoadData(self, layer, columns=None, where=None, sql=None):
 
-            record = record.split(fs)
-            if len(columns) != len(record):
-                # Assuming there will be always at least one.
-                last = record[-1]
-                show_max = 3
-                if len(record) > show_max:
-                    record = record[:show_max]
-                # TODO: The real fix here is to use JSON output from v.db.select or
-                # proper CSV output and real CSV reader here (Python csv and json
-                # packages).
-                raise GException(
-                    _(
-                        "Unable to read the table <{table}> from the database due"
-                        " to seemingly inconsistent number of columns in the data"
-                        " transfer."
-                        " Check row: {row}..."
-                        " Likely, a newline character is present in the attribute value"
-                        " starting with: '{value}'"
-                        " Use the v.db.select module to investigate."
-                    ).format(table=tableName, row=" | ".join(record), value=last)
-                )
-                self.columns = {}  # because of IsEmpty method
-                return None
+                record = record.split(fs)
+                if len(columns) != len(record):
+                    # Assuming there will be always at least one.
+                    last = record[-1]
+                    show_max = 3
+                    if len(record) > show_max:
+                        record = record[:show_max]
+                    # TODO: The real fix here is to use JSON output from v.db.select or
+                    # proper CSV output and real CSV reader here (Python csv and json
+                    # packages).
+                    raise GException(
+                        _(
+                            "Unable to read the table <{table}> from the database due"
+                            " to seemingly inconsistent number of columns in the data"
+                            " transfer."
+                            " Check row: {row}..."
+                            " Likely, a newline character is present in the attribute value"
+                            " starting with: '{value}'"
+                            " Use the v.db.select module to investigate."
+                        ).format(table=tableName, row=" | ".join(record), value=last)
+                    )
+                    self.columns = {}  # because of IsEmpty method
+                    return None
 
-            self.AddDataRow(i, record, columns, keyId)
+                self.AddDataRow(i, record, columns, keyId)
 
-            i += 1
-            if i >= 100000:
-                self.log.write(_("Viewing limit: 100000 records."))
-                break
+                i += 1
+                if i >= 100000:
+                    self.log.write(_("Viewing limit: 100000 records."))
+                    break
 
-        self.SetItemCount(i)
```

Comment:
At this point the file can be closed, no need to include the rest in the context manager, no?